common/src/main/java/io/netty/util/ThreadDeathWatcher.java

```diff
@@ -137,13 +129,10 @@ public Void run() {
      * @return {@code true} if and only if the watcher thread has been terminated
      */
     public static boolean awaitInactivity(long timeout, TimeUnit unit) throws InterruptedException {
-        if (unit == null) {
-            throw new NullPointerException("unit");
-        }
-
         Thread watcherThread = ThreadDeathWatcher.watcherThread;
         if (watcherThread != null) {
-            watcherThread.join(unit.toMillis(timeout));
+            watcherThread.join(
+                    ObjectUtil.checkNotNull(unit, "unit").toMillis(timeout));
```

Comment:
move the null check as first action in the method to ensure we also throw if `watcherThread == null` and `unit` is null